homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -149,14 +169,20 @@ async def async_update(self):
                 )
         else:
             when = datetime.now()
-            self._state = await self._train_api.async_get_next_train_stop(
-                self._from_station, self._to_station, when
-            )
-        self._departure_state = self._state.get_state().name
-        self._delay_in_minutes = self._state.get_delay_time()
+            try:
+                self._state = await self._train_api.async_get_next_train_stop(
+                    self._from_station, self._to_station, when
+                )
+            except ValueError as output_error:
+                _LOGGER.error(
+                    "Departure %s encountered a problem: %s", when, output_error
+                )
+        if self._state:
+            self._departure_state = self._state.get_state().name
+            self._delay_in_minutes = self._state.get_delay_time()
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, Any] | None:
```

Comment:
Could you go one step further and simply set `self._attr_extra_state_attributes` inside `async_update`?